### DISCUSSION TITLE
Heltec V3

### DISCUSSION THREAD

**Discussion by dkallen78** [dl9rdz/rdz_ttgo_sonde#424] [Q&A]
Are there any plans to update this for the Heltec V3 with the SX1262 chip? What could one do to help facilitate that? Thanks!

**Comment by dl9rdz** [comment]
Not right now. My experiments with the chip were somewhat disappointing.

It is a completely different chip, capable of doing much more high level stuff (which is not compatible with decoding radiosondes), but not with a great API for doing low-level stuff like on the SX127x chips. 

So far I am not yet able to tell if it is at all able to tune on 40x MHz (this is outside of the specs of the chip). I tried sending some commands to the chip, and I am getting more or less random replies (errors and non-errors).

What one can do: A proof-of-concept implementation showing that the chip is indeed capable of receiving *some* FSK signal on 400 MHz. If I get that I will be more confident that receiving some radiosonde is actually feasible.

**Reply by buttim** [reply]
Actually the sx1262 is capable of tuning in the 400MHz range since its complete range goes from 150 to 960MHz. Not to be confused with the sx1268 that has a 410MHz lower limit. Still it is not clear if it can be used for radiosondes. It has a limitation of 255 bytes for the payload that can be overcome with some hacks (see [this application note](https://semtech.my.salesforce.com/sfc/p/E0000000JelG/a/3n000000qSr5/_k7DWETSW27pcUXl2q6mgqTRbGdaR02ue716zGGCLMQ)).
I'm struggling to get some result but currently stuck with receiving the wrong bytes from an RS41

**Reply by buttim** [reply]
I finally solved my problem. The order of the bits is reversed, so I had to flip every byte both in the sync word programming and in the data received. It actually should be feasible to receive an RS41, but I didn't test the hack for receiving packets longer than 255 bytes. I'm quite confident it will work though. I hope I will be able to give you the good news in a few days

**Reply by buttim** [reply]
Long packets seem to work, but not reliably for the moment. I checked the first received packet and it was 100% correct. Well, at least all the blocks had a correct CRC.
The code is [here](https://github.com/buttim/provaSX1262)
I'm going to make some more tests an hopefully make it fully reliable.
Please let me know if you have a chance to give it a look

**Reply by buttim** [reply]
WARNING: I'm not using a ~~TBeam~~ Heltec V3 but an ESP32 Devkit with an external SX1262 module.

**Reply by dl9rdz** [reply]
I will have a look at it when I am back home (in two days). Looks interesting.

So at least RS41 should then be feasible.  For all other types, the SX1278 is used without using its preamble detector, I wonder if this is doable with the SX1262...

**Reply by buttim** [reply]
I wasn't aware of that. You can exclude the preamble detector by specifying
`.preamble_detector = SX126X_GFSK_PREAMBLE_DETECTOR_OFF`
in the declaration of the `pktParams` variable, according to the datasheet.
I will add a minimal README to the project in the next days

**Reply by buttim** [reply]
Any news?

**Reply by TheFilthyFrog** [reply]
Any updates

**Comment by buttim** [comment]
2 years and not even a negative response. So much for the 2 days.
If you are just wanting to do on the field hunting for sondes I wrote [this firmware](https://www.ydiaeresis.eu/public/index.html) which works **both** for the Heltec v3 and the TTGO LoRa32.
It uses a different approach, using bluetooth instead of wifi. To use it you will need the ["Trova La Sonda" Android app](https://play.google.com/store/apps/details?id=eu.ydiaeresis.trovalasonda).
